src/community/geogig/src/main/java/org/geogig/geoserver/rest/AddRepoRequestHandler.java

```diff
@@ -27,18 +23,26 @@
 import org.restlet.ext.json.JsonRepresentation;
 import org.restlet.resource.Representation;
 
-import com.google.common.annotations.VisibleForTesting;
-import com.google.common.base.Optional;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.locationtech.geogig.web.api.RESTUtils.getStringAttribute;
+import static org.restlet.data.Status.CLIENT_ERROR_BAD_REQUEST;
 
 /**
  * Utility for handling GeoGIG repository init requests. This class will pull repository creation
  * details (like parent directory, or PostgreSQL database connection parameters) from the Request
  * and build a GeoGIG repository form them, by converting the request into a
  * {@link org.locationtech.geogig.repository.Hints Hints}.
  */
-class InitRequestHandler {
+class AddRepoRequestHandler {
```

Comment:
This class was refactored out of the plugin, so the updates to support specifying a leaf directory will have to be made on the geogig side while the importGeogig function can be moved.  (See the recent changes in the plugin and model it after the init changes).